packages/opentelemetry-api/src/api/global-utils.ts

```diff
@@ -15,54 +15,74 @@
 
-type Get<T> = (version: number) => T;
-type OtelGlobal = Partial<{
-  [GLOBAL_CONTEXT_MANAGER_API_KEY]: Get<ContextManager>;
-  [GLOBAL_METRICS_API_KEY]: Get<MeterProvider>;
-  [GLOBAL_PROPAGATION_API_KEY]: Get<TextMapPropagator>;
-  [GLOBAL_TRACE_API_KEY]: Get<TracerProvider>;
-}>;
+export function registerGlobal(type: 'trace', instance: TracerProvider): void;
+export function registerGlobal(type: 'metrics', instance: MeterProvider): void;
+export function registerGlobal(type: 'context', instance: ContextManager): void;
+export function registerGlobal(
+  type: 'propagation',
+  instance: TextMapPropagator
+): void;
+export function registerGlobal(type: keyof OTelGlobalApi, instance: any) {
+  _global[GLOBAL_OPENTELEMETRY_API_KEY] =
+    _global[GLOBAL_OPENTELEMETRY_API_KEY] ?? {};
 
-export const _global = _globalThis as OtelGlobal;
+  const api = _global[GLOBAL_OPENTELEMETRY_API_KEY]!;
+  if (api[type]) {
+    // already registered an API of this type
+    return;
+  }
 
-/**
- * Make a function which accepts a version integer and returns the instance of an API if the version
- * is compatible, or a fallback version (usually NOOP) if it is not.
- *
- * @param requiredVersion Backwards compatibility version which is required to return the instance
- * @param instance Instance which should be returned if the required version is compatible
- * @param fallback Fallback instance, usually NOOP, which will be returned if the required version is not compatible
- */
-export function makeGetter<T>(
-  requiredVersion: number,
-  instance: T,
-  fallback: T
-): Get<T> {
-  return (version: number): T =>
-    version === requiredVersion ? instance : fallback;
+  api[type] = {
+    instance,
+    version: VERSION,
+  };
 }
 
-/**
- * A number which should be incremented each time a backwards incompatible
- * change is made to the API. This number is used when an API package
- * attempts to access the global API to ensure it is getting a compatible
- * version. If the global API is not compatible with the API package
- * attempting to get it, a NOOP API implementation will be returned.
- */
-export const API_BACKWARDS_COMPATIBILITY_VERSION = 3;
+export function getGlobal(type: 'trace'): Signal<TracerProvider> | undefined;
+export function getGlobal(type: 'metrics'): Signal<MeterProvider> | undefined;
+export function getGlobal(type: 'context'): Signal<ContextManager> | undefined;
+export function getGlobal(
+  type: 'propagation'
+): Signal<TextMapPropagator> | undefined;
+export function getGlobal(type: keyof OTelGlobalApi) {
+  return _global[GLOBAL_OPENTELEMETRY_API_KEY]?.[type];
+}
+
+export function unregisterGlobal(type: keyof OTelGlobalApi) {
+  const api = _global[GLOBAL_OPENTELEMETRY_API_KEY];
+
+  if (api) {
+    delete api[type];
+  }
+}
+
+export function isCompatible(version: string) {
+  return semver.satisfies(version, acceptableRange);
+}
+
+type OTelGlobal = Partial<{
+  [GLOBAL_OPENTELEMETRY_API_KEY]: OTelGlobalApi;
```

Comment:
```suggestion
  [GLOBAL_OPENTELEMETRY_API_KEY]?: OTelGlobalApi;
```
and remove the `Partial<`